application_form/api/sales/views.py

```diff
@@ -69,18 +69,30 @@ class ApartmentReservationViewSet(mixins.RetrieveModelMixin, viewsets.GenericVie
     permission_classes = [permissions.AllowAny]
 
     @extend_schema(
-        description="Create a HASO contract PDF for the reservation.",
+        description="Create either a Hitas contract or a HASO contract PDF based on "
+        "the reservation's project's ownership type.",
         responses={(200, "application/pdf"): OpenApiTypes.BINARY},
     )
     @action(methods=["GET"], detail=True)
-    def haso_contract(self, request, pk=None):
+    def contract(self, request, pk=None):
         reservation = self.get_object()
-        pdf_data = create_haso_contract_pdf(reservation)
-
-        apartment = get_apartment(reservation.apartment_uuid)
+        apartment = get_apartment(
+            reservation.apartment_uuid, include_project_fields=True
+        )
         title = (apartment.title or "").strip().lower().replace(" ", "_")
-        filename = f"haso_sopimus_{title}.pdf" if title else "haso_sopimus.pdf"
+
+        if apartment.project_ownership_type == "Hitas":
+            filename = f"hitas_sopimus_{title}" if title else "hitas_sopimus"
+            pdf_data = create_hitas_contract_pdf(reservation)
+        elif apartment.project_ownership_type == "HASO":
+            filename = f"haso_sopimus_{title}" if title else "haso_sopimus"
```

Comment:
Are these filename uniques if I want to download multiple contracts?

Reply:
If multiple contracts are loaded for the same apartment, then the resulting filename will be the same. However, the sales UI doesn't actually even use these filenames, so that shouldn't be an actual problem in any case 🙂